system-config/setup-ccache-and-icecc.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/bash
+# vagrant-cookbook
+# Copyright (C) 2015-2017 Pelagicore AB
+#
+# Permission to use, copy, modify, and/or distribute this software for
+# any purpose with or without fee is hereby granted, provided that the
+# above copyright notice and this permission notice appear in all copies.
+#
+# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL
+# WARRANTIES WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR
+# BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES
+# OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+# WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+# ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS
+# SOFTWARE.
+#
+# For further information see LICENSE
+#
+#
+# Usage: setup-ccache-and-icecc.sh
+#
+# Installs and configures ccache and icecc
+#
+
+function install {
+    packages="$@"
+
+    retval=100
+    count=0
+
+    if [[ "$retval" -ne "0" && $count -le 5 ]]; then
+        count=$count+1
+        DEBIAN_FRONTEND=noninteractive apt-get -o Dpkg::Options::="--force-confnew" --force-yes -fuy install $packages
+        retval=$?
+    fi
+
+    if [[ "$retval" -ne "0" ]]; then
+        echo "Failed to install " $packages
+        exit $retval
+    fi
+
+}
+
+install ccache icecc
+
+echo 'export CC="ccache icecc gcc"' >> /etc/profile
+echo 'export CXX="ccache icecc g++"' >> /etc/profile
```

Comment:
As we discussed offline, I do not like the idea of forcing the use of gcc. Bind to `/usr/bin/cc` and `/usr/bin/c++` if they exist and default to gcc and g++ otherwise. Even though Clang seems to work poorly with icecc this solution atleast enables the possibility.